### ISSUE TITLE
Add ShowWeek option in DatePicker

### ISSUE THREAD

**Issue by yuxinbo** [ant-design/ant-design#21726]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
For some company, most of works are arranged by week. Through the detail work will be linked with one day, but they hope to know which week the current selected date belong to. This is really wonderful feature as most calendar component miss this feature.

### What does the proposed API look like?
Hope to add 'ShowWeek' option for DatePicker, when this options is selected, show one week column on the left of current calendar as WeekPicker. (Looks like WeekPicker but is able to select date).

**Comment by GoogleAppEngine** [comment]
+1
I want this feature as well

**Comment by KingGoujian** [comment]
We want this feature too.

**Comment by nguyenhoangnamkg** [comment]
I want this feature too.

**Comment by andersjonsson** [comment]
+1
I want this feature as well

**Comment by jtnidrizi** [comment]
+1
we need this as well

**Comment by f-do** [comment]
+1

**Comment by vsosinga** [comment]
Would like this feature as well

**Comment by hoest** [comment]
+1

**Comment by yuxincheng73** [comment]
Reopening this issue.

I would like to work on this issue but I suppose I would need to make changes to [react-component/picker](https://github.com/react-component/picker?fbclid=IwAR3MHaRiEmAE8F3LqkgAis0CiOtqviX0HUW0fv7j1XMn9Det3hkbVD2e0jM) and ant-design/date-picker? If that is the case, will I need to wait for the PR to be approved in the react-component/picker project first and then proceed to updating ant-design/date-picker, since rc-picker is a imported dependency?

**Comment by f-do** [comment]
Yes correct, the option like "showCalendarWeek" must first implemented in the picker, the Option is more or less already there. If you select a full calendar week the week number is visible. It is just one attribute. After that ant can also access this attribute.

**Comment by yuxincheng73** [comment]
I made two a PR to [react-component/picker](https://github.com/react-component/picker/pull/326) and to this repository. This is what it looks like: 
![ezgif com-gif-maker (6)](https://user-images.githubusercontent.com/50343180/146267939-d468e645-4e6b-4073-829c-cda06ea435b7.gif)

I tested locally by npm publishing a copy of the edited rc-picker but for this to work in general, that PR must be approved first, or else the changes to rc-picker aren't present in the node package.

**Comment by myrdrks** [comment]
+1

**Comment by Vagizova** [comment]
+1

**Comment by andrei-taras** [comment]
+1
I want this feature as well.

**Comment by vsosinga** [comment]
@zombieJ @afc163 what else is needed to add this feature?

**Comment by vsosinga** [comment]
@afc163 a PR for [react-component/picker](https://github.com/react-component/picker/pull/326) was already made by @yuxincheng73 . This has to be merged first before we can create a pull request for ant design.

**Comment by vsosinga** [comment]
@yuxincheng73 thanks for your effort, do you have time to update your PR's and fix the conflicts?

**Comment by stijnvanhulle** [comment]
What is the status of this, we want to use a date picker and show the week numbers but we have not found a way of doing this with the current setup of the date-picker. `showWeekNumbers=false` would resolve this.

**Comment by stijnvanhulle** [comment]
@zombieJ It seems this has been added in v5.14.0, so this issue can probably be closed?

**Comment by zombieJ** [comment]
Picker component has been refactor in `5.14.0`, and this issue is discussed and handled with https://github.com/react-component/picker/pull/691 which will be marked as closed.

Please feel free to create new issue if you have any further question :)

Picker 组件于 `5.14.0` 版本进行了重构，此问题已被 https://github.com/react-component/picker/pull/691 中进行讨论与处理。

如果有任何问题，欢迎随时创建新的 issue :)
